src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -11,6 +11,23 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
+    internal enum NullableAnnotation : byte
+    {
+        Unknown,     // No information. Think oblivious.
+        NotNullable, // For string, int, T
+        Nullable,    // For string?, T? where T : class; and for int?, T? where T : struct.
+        NotNullableBasedOnAnalysis, // Explicitly set by flow analysis
+        NullableBasedOnAnalysis, // Explicitly set by flow analysis
```

Comment:
> Could the Analysis part be a separate bit?

I thought about that, but for now decided against using flags to avoid the need to deal with "not meaningful"/impossible combinations. Also, I am hoping to get rid of NullableBasedOnAnalysis in the next change, making all values fit in two bits. 

> Or a separate bool field on TypeSymbolWithAnnotations?

I find it more convenient to deal with single value than with a pair of values. Going forward we should be able to get rid of Create methods that don't take NullableAnnotation.


---
In reply to: [230184941](https://github.com/dotnet/roslyn/pull/30913#discussion_r230184941) [](ancestors = 230184941)

Reply:
> I find it more convenient to deal with single value than with a pair of values.

Which callers are treating the pair as one value?

---
In reply to: [230193197](https://github.com/dotnet/roslyn/pull/30913#discussion_r230193197) [](ancestors = 230193197,230184941)

Reply:
> Which callers are treating the pair as one value?

I think every caller as it stands right now, the TypeSymbolWithAnnotations itself treats it as one value when merging transitioning between annotations. With pair of values it is not clear what combinations the code should be prepared to deal with. 

---
In reply to: [230195433](https://github.com/dotnet/roslyn/pull/30913#discussion_r230195433) [](ancestors = 230195433,230193197,230184941)